docs/architecture.md

```diff
@@ -391,6 +391,34 @@ type ValidatorAPI interface {
 }
 ```
 
+### Signer
+The signer provides support the alternative _Remote Signature_ architecture
+which authorises charon to request adhoc signatures from a [remote signer instance](https://lighthouse-book.sigmaprime.io/validator-web3signer.html).
+In the _middleware_ architecture charon cannot initiate signatures itself and has to
+wait for the VC to submit signatures.
+
+Duties originating in the `scheduler` (`DutyAttester`, `DutyProposer`) are not significantly affected by this change in architecture.
+Instead of waiting for the `validatorapi` to submit signatures, these duties directly request
+signatures from the remote signer instance. The flow is otherwise unaffected.
+
+Duties originating in the `validatorapi` (`DutyRandao`, `DutyAggregator`) has to refactored to
+originate in the `scheduler`, since charon is in full control of the duties in this architecture.
+
+The overall core workflow remains the same, `scheduler` just schedules all the duties.
+
+> 🏗️ TODO: Figure out if signer should query DutyDB for slashing, or if DutyDB should push to signer.
+
+```go
+// Signer signs unsigned duty data sets via one or more remote signer instances.
+type Signer interface {
+    // Sign signs the unsigned duty data set.
+    Sign(context.Context, Duty, UnsignedDataSet) error
+
+    // RegisterParSigDB registers a function to store partially signed data sets.
+    RegisterParSigDB(func(context.Context, Duty, ParSignedDataSet) error))
```

Comment:
Mandatory dependencies should be in the constructors. Not in setters. In most Go code, including Prysm we use interfaces instead of function types for dependencies

Reply:
So this is the architecture we are following. We wire components via input and output functions.